docs/page-objects/inputs.rst

```diff
@@ -0,0 +1,51 @@
+.. _inputs:
+
+======
+Inputs
+======
+
+:ref:`Page object classes <page-object-classes>`, in their ``__init__`` method,
+must define input parameters with type hints pointing to classes from the
+:mod:`web_poet.page_inputs` module.
```

Comment:
As you mention later, it could be custom classes as well.

Also, one may use other injectable things as dependencies: other page objects, items (given that the framework implements providing items).